include/dali/core/tensor_shape.h

```diff
@@ -637,6 +638,43 @@ struct TensorListShapeBase {
     shapes.resize(num_samples * sample_dim);
   }
 
+
+  ///@{
+  /**
+   * Append this TensorListShape with shapes from other TensorListShapes.
+   */
+  void append(span<const TensorListShape<sample_ndim>> tlss) {
+    int new_samples = std::accumulate(tlss.begin(), tlss.end(), 0,
+                                      [](int acc, const auto &tls) -> int {
+                                          return acc + tls.num_samples();
+                                      });
+    if (new_samples == 0) return;
+    for (const auto &tls : tlss) {
+      if (tls.sample_dim() != tlss[0].sample_dim()) {
+        throw std::invalid_argument(
```

Comment:
shouldn't you check that the number of dimensions is also the same as the current instance you are appending to?

Reply:
Fixed